qa/python_models/bls_model_loading/model.py

```diff
@@ -0,0 +1,119 @@
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import numpy as np
+import unittest
+import triton_python_backend_utils as pb_utils
+
+
+class PBBLSModelLoadingTest(unittest.TestCase):
+
+    def setUp(self):
+        self.model_name = "onnx_int32_int32_int32"
+
+    def tearDown(self):
+        pb_utils.unload_model(self.model_name)
+
+    def test_load_unload_model(self):
+        self.assertFalse(pb_utils.is_model_ready(model_name=self.model_name))
+        pb_utils.load_model(model_name=self.model_name)
+        self.assertTrue(pb_utils.is_model_ready(self.model_name))
+        pb_utils.unload_model(self.model_name)
+        self.assertFalse(pb_utils.is_model_ready(self.model_name))
+
+    def test_load_with_config_override(self):
+        self.assertFalse(pb_utils.is_model_ready(self.model_name))
+        pb_utils.load_model(self.model_name)
+        self.assertTrue(pb_utils.is_model_ready(self.model_name))
+
+        # Send the config with the wrong format
+        wrong_config = "\"parameters\": {\"config\": {{\"backend\":\"onnxruntime\", \"version_policy\":{\"specific\":{\"versions\":[2]}}}}}"
+        with self.assertRaises(pb_utils.TritonModelException):
+            pb_utils.load_model(model_name=self.model_name, config=wrong_config)
+        # The model should not be changed after a failed load model request
+        for version in ["2", "3"]:
+            self.assertTrue(pb_utils.is_model_ready(model_name=self.model_name, model_version=version))
+
+        # Send the config with the correct format
+        config = "{\"backend\":\"onnxruntime\", \"version_policy\":{\"specific\":{\"versions\":[2]}}}"
+        pb_utils.load_model(self.model_name, config=config)
+        # The model should be changed after a successful load model request
+        self.assertTrue(pb_utils.is_model_ready(self.model_name, "2"))
+        self.assertFalse(pb_utils.is_model_ready(self.model_name, "3"))
+
+    def test_load_with_file_override(self):
+        self.assertFalse(pb_utils.is_model_ready(self.model_name))
+        pb_utils.load_model(self.model_name)
+        self.assertTrue(pb_utils.is_model_ready(self.model_name))
+
+        override_name = "override_model"
+        config = "{\"backend\":\"onnxruntime\"}"
+        with open('models/onnx_int32_int32_int32/3/model.onnx', 'rb') as file:
+            data = file.read()
+        files = {"file:1/model.onnx": data}
+
+        # Request to load the model with override file, should fail without
+        # providing override config.
+        with self.assertRaises(pb_utils.TritonModelException):
+            pb_utils.load_model(self.model_name, "", files)
```

Comment:
I was wondering why would it fail if config is not provided?

Reply:
This was from the core logic: https://github.com/triton-inference-server/core/blob/51674037a179fa661ae53b42c8d2bfcbf64a425a/src/model_repository_manager.cc#L145